### PR TITLE
Changed Defaults to Tools

### LINKED ISSUE #679: `DeprecationWarning` raised on `import instructor` vs. use of `Mode.FUNCTIONS`
- [X] This is actually a bug report.
- ~I am not getting good LLM Results
- ~I have tried asking for help in the community on discord or discussions and have not received a response.~
- ~I have tried searching the documentation and have not found an answer.~

**What Model are you using?**
N/A

**Describe the bug**
Instructor is raising a `DeprecationWarning` on import vs. access of the `Mode.FUNCTIONS` mode.

**To Reproduce**
```python3
import warnings
warnings.simplefilter("default")
import instructor
```

**Expected behavior**
A `DeprecationWarning` is only raised on access to `Mode.FUNCTIONS`

**Screenshots**
N/A

### PR DESCRIPTION
**Pull request by ivanleomk**
This should fix #679 

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 42067ea91fdd38248b07d8596cf57c280ff1c0f5  | 
|--------|--------|

### Summary:
Removed deprecated `Mode.FUNCTIONS`, updated references to `Mode.TOOLS`, and added a warning for any access to `Mode.FUNCTIONS`.

**Key points**:
- Removed deprecated `Mode.FUNCTIONS`.
- Updated references from `Mode.FUNCTIONS` to `Mode.TOOLS` in `docs/concepts/patching.md`, `examples/logging/run.py`, `examples/vision/image_to_ad_copy.py`.
- Modified `instructor/client.py`, `instructor/dsl/iterable.py`, `instructor/dsl/partial.py`, `instructor/function_calls.py`, `instructor/mode.py`, `instructor/process_response.py` to remove `Mode.FUNCTIONS` and related logic.
- Updated tests in `tests/test_function_calls.py` to use `Mode.TOOLS`.
- Added a warning for any access to `Mode.FUNCTIONS` in `instructor/mode.py`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: instructor/dsl/partial.py [hunk 1]
```diff
@@ -191,10 +188,7 @@ async def extract_json_async(
                 if mode == Mode.ANTHROPIC_TOOLS:
                     yield chunk.delta.partial_json
                 elif chunk.choices:
-                    if mode == Mode.FUNCTIONS:
```
Comment (jxnl) [inline instructor/dsl/partial.py:194]: lets not delete functions, since older openai openapi wrappers might use functions vs tools

**Review by jxnl — APPROVED** [review]
dont delete functions yet, old openai wrapper might use one or the other.

just change defaults

**Comment by intellectronica** [comment]
https://github.com/jxnl/instructor/pull/863 retains the warning, but only triggers it when actually using FUNCTIONS, not when reading the value.

**Comment by ivanleomk** [comment]
@jxnl I changed the defaults and reverted the changes otherwise.

**Pull request merged**